projects/saas-ui/src/app/main/components/manage-plans/manage-plans.component.ts

```diff
@@ -53,18 +57,48 @@ export class ManagePlansComponent
     private fb: FormBuilder,
   ) {
     super(route, location);
+    this.gridOptions = {
+      pagination: true,
+      rowModelType: 'infinite',
+      paginationPageSize: this.limit,
+      paginationPageSizeSelector: [this.limit, 10, 20, 50, 100],
+      cacheBlockSize: this.limit,
+      onGridReady: this.onGridReady.bind(this),
+      rowHeight: 60,
+      defaultColDef: {flex: 1},
+    };
   }
 
-  ngOnInit(): void {
-    this.getPlans();
+  onGridReady(params: AnyObject) {
+    this.gridApi = params.api;
+    const dataSource: IDatasource = {
+      getRows: (params: IGetRowsParams) => {
+        const page = params.endRow / this.limit;
+        const paginatedLeads = this.getPaginatedPlans(page, this.limit);
+        const totalLead = this.getTotal();
+        combineLatest([paginatedLeads, totalLead]).subscribe(
+          ([data, count]) => {
+            params.successCallback(data, count.count);
+          },
+
+          err => {
+            params.failCallback();
+          },
+        );
+      },
+    };
+    params.api.setDatasource(dataSource);
   }
 
-  getPlans() {
-    this.billingPlanService
-      .getPlanOptions(null, null, this.filter)
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(res => {
-        this.rowData = res.map(item => {
+  getPaginatedPlans(page: number, limit: number): Observable<any[]> {
```

Comment:
again same 
please check all files and remove

Reply:
sorry removed